# Titanic Wrangling

In this practice activity you'll continue to work with the titanic dataset in ways that flex what you've learned about both data wrangling and data visualization.

In [6]:
import numpy as np
import pandas as pd

# Import the titanic dataset here
data_dir = "https://dlsun.github.io/pods/data/"
df_titanic = pd.read_csv(data_dir + "titanic.csv")
df_titanic

name  gender   age             class embarked  \
0                Abbing, Mr. Anthony    male  42.0               3rd        S   
1          Abbott, Mr. Eugene Joseph    male  13.0               3rd        S   
2        Abbott, Mr. Rossmore Edward    male  16.0               3rd        S   
3     Abbott, Mrs. Rhoda Mary 'Rosa'  female  39.0               3rd        S   
4        Abelseth, Miss. Karen Marie  female  16.0               3rd        S   
...                              ...     ...   ...               ...      ...   
2202                Wynn, Mr. Walter    male  41.0         deck crew        B   
2203             Yearsley, Mr. Harry    male  40.0  victualling crew        S   
2204        Young, Mr. Francis James    male  32.0  engineering crew        S   
2205             Zanetti, Sig. Minio    male  20.0  restaurant staff        S   
2206              Zarracchi, Sig. L.    male  26.0  restaurant staff        S   

            country  ticketno   fare  survived  
0     United States    5547.0   7.11         0  
1     United States    2673.0  20.05         0  
2     United States    2673.0  20.05         0  
3           England    2673.0  20.05         1  
4            Norway  348125.0   7.13         1  
...             ...       ...    ...       ...  
2202        England       NaN    NaN         1  
2203        England       NaN    NaN         1  
2204        England       NaN    NaN         0  
2205        England       NaN    NaN         0  
2206        England       NaN    NaN         0  

[2207 rows x 9 columns]

## 1. Filter the data to include passengers only. Calculate the joint distribution (cross-tab) between a passenger's class and where they embarked.

In [18]:
# Code here
passengerdata = df_titanic[(df_titanic["class"] == "1st") | (df_titanic["class"] == "2nd") | (df_titanic["class"] == "3rd")]
passengerdata

name  gender   age class embarked  \
0                Abbing, Mr. Anthony    male  42.0   3rd        S   
1          Abbott, Mr. Eugene Joseph    male  13.0   3rd        S   
2        Abbott, Mr. Rossmore Edward    male  16.0   3rd        S   
3     Abbott, Mrs. Rhoda Mary 'Rosa'  female  39.0   3rd        S   
4        Abelseth, Miss. Karen Marie  female  16.0   3rd        S   
...                              ...     ...   ...   ...      ...   
1314          Yvois, Miss. Henriette  female  24.0   2nd        S   
1315       Zakarian, Mr. Mapriededer    male  22.0   3rd        C   
1316             Zakarian, Mr. Ortin    male  27.0   3rd        C   
1317               Zenni, Mr. Philip    male  25.0   3rd        C   
1318             Zimmermann, Mr. Leo    male  29.0   3rd        S   

            country  ticketno     fare  survived  
0     United States    5547.0   7.1100         0  
1     United States    2673.0  20.0500         0  
2     United States    2673.0  20.0500         0  
3           England    2673.0  20.0500         1  
4            Norway  348125.0   7.1300         1  
...             ...       ...      ...       ...  
1314         France  248747.0  13.0000         0  
1315         Turkey    2656.0   7.0406         0  
1316         Turkey    2670.0   7.0406         0  
1317        Lebanon    2620.0   7.0406         1  
1318        Germany  315082.0   7.1706         0  

[1317 rows x 9 columns]

In [28]:
pd.crosstab(passengerdata["embarked"], passengerdata["class"])

class     1st  2nd  3rd
embarked               
B           3    6    0
C         143   26  102
Q           3    7  113
S         175  245  494

## 2. Using the joint distribution that calculated above, calculate the following:

* the conditional distribution of their class given where they embarked
* the conditional distribution of where they embarked given their class

Use the conditional distributions that you calculate to answer the following quesitons:

* What proportion of 3rd class passengers embarked at Southampton?
* What proportion of Southampton passengers were in 3rd class?

In [31]:
# Code here
pd.crosstab(passengerdata["embarked"], passengerdata["class"], normalize = "index")

class          1st       2nd       3rd
embarked                              
B         0.333333  0.666667  0.000000
C         0.527675  0.095941  0.376384
Q         0.024390  0.056911  0.918699
S         0.191466  0.268053  0.540481

In [32]:
pd.crosstab(passengerdata["class"], passengerdata["embarked"], normalize = "index")

embarked         B         C         Q         S
class                                           
1st       0.009259  0.441358  0.009259  0.540123
2nd       0.021127  0.091549  0.024648  0.862676
3rd       0.000000  0.143865  0.159379  0.696756

69.68% of all 3rd class passengers embarked at Southampton.

54.05% of Southampton passengers were in 3rd class.

## 3. Make a visualization showing the distribution of a passenger's class, given where they embarked.

Discuss the pros and cons of using this visualization versus the distributions you calculated before, to answer the previous questions.

In [33]:
# Code here
pd.crosstab(passengerdata["embarked"], passengerdata["class"] == "1st", normalize = "index")

class        False     True 
embarked                    
B         0.666667  0.333333
C         0.472325  0.527675
Q         0.975610  0.024390
S         0.808534  0.191466

In [35]:
pd.crosstab(passengerdata["embarked"], passengerdata["class"] == "2nd", normalize = "index")

class        False     True 
embarked                    
B         0.333333  0.666667
C         0.904059  0.095941
Q         0.943089  0.056911
S         0.731947  0.268053

In [36]:
pd.crosstab(passengerdata["embarked"], passengerdata["class"] == "3rd", normalize = "index")

class        False     True 
embarked                    
B         1.000000  0.000000
C         0.623616  0.376384
Q         0.081301  0.918699
S         0.459519  0.540481

Pros of using crosstab given condition
- easier to understand if no knowledge of conditional distirbution
Cons of using crosstab given condition
- more coding required, False and True may be confusing